crates/parser/src/ix_parser/token_extensions/token_extension_parser.rs

```diff
@@ -0,0 +1,117 @@
+use helpers::ExtensionIxParser;
+use spl_token_2022::instruction::TokenInstruction;
+use spl_token_group_interface::instruction::TokenGroupInstruction;
+use spl_token_metadata_interface::instruction::TokenMetadataInstruction;
+
+use super::{extensions::*, token_extensions_ix::TokenExtensionProgramIx};
+use crate::ix_parser::{
+    token_program::TokenProgramIxParser,
+    vixen_ix::structure::{InstructionParser, InstructionUpdate},
+};
+
+#[derive(Debug)]
+pub struct TokenExtensionProgramIxParser;
+
+impl InstructionParser<TokenExtensionProgramIx> for TokenExtensionProgramIxParser {
+    fn parse_readable_ix(ix_update: &InstructionUpdate) -> Result<TokenExtensionProgramIx, String> {
+        match TokenInstruction::unpack(&ix_update.data) {
+            Ok(ix) => match ix {
```

Comment:
This is looking good. Nj. 
```suggestion
impl InstructionParser<TokenExtensionProgramIx> for TokenExtensionProgramIxParser {
    fn parse(ix_update: &InstructionUpdate) -> Result<TokenExtensionProgramIx, String> {
        match TokenInstruction::unpack(&ix_update.data) {
            Ok(ix) => match ix {
```

Reply:
renaming to parse_ix bcos we already have parse fn inside Parser trait it will cause fn name conflict